### ISSUE TITLE
Truthy/Falsey values don't do what yu

### ISSUE THREAD

**Issue by ejether** [reactiveops/autohelm#7]
extraArgs.skip-nodes-with-system-pods\=false: ""

**Comment by ejether** [comment]
is an issue with how the `oyaml` parses truthy values to `oyaml` the strings `"true"`, `True`, `'True'` etc all get parsed to the python boolean `True`

**Comment by endzyme** [comment]
Here's the rundown on the issue. It basically boils down to a loss of typing fidelity when converting elements in YAML to command line arguments. 

When you set `values: {thing: "true"}` or `values-strings: {thing: "true"}` in the autohelm course, these values are put into `--set=thing=true` and `--set-string=thing=true` respectively. Unfortunately helm is [working as intended](https://github.com/helm/helm/issues/4006) when it takes the `--set-string=thing=true` and makes the YAML value into `values: {thing: true}` (note lack of quotes, so bool type instead of string).

The only way to get around this is to avoid using command line argument value setting, which means using the `files: []` config and providing autohelm a file to use as the Values.yaml for the chart install. This keeps the typing correct for elements like `true/false`. 

TL;DR - `--set-string=X` only is intended to work on large integers for yaml safety, NOT bool types.

**Comment by endzyme** [comment]
Also this issue comes up quickly when you are trying to use a template that uses `toYaml` in a `kind: ConfigMap` for `data: {}` -- Since the schema only accepts strings as values things lose the type fidelity and break. :man_facepalming:
